controllers/quay/quayregistry_controller.go

```diff
@@ -345,68 +419,8 @@ func (r *QuayRegistryReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		updatedQuay, err = r.updateWithCondition(updatedQuay, v1.ConditionComponentsCreated, metav1.ConditionFalse, v1.ConditionReasonMigrationsInProgress, "running database migrations")
 		if err != nil {
 			log.Error(err, "failed to update `conditions` of `QuayRegistry`")
-
-			return ctrl.Result{}, nil
 		}
-
-		go func(quayRegistry *v1.QuayRegistry) {
-			err = wait.Poll(upgradePollInterval, upgradePollTimeout, func() (bool, error) {
-				log.Info("checking Quay upgrade `Job` completion")
-
-				var upgradeJob batchv1.Job
-				err = r.Client.Get(ctx, types.NamespacedName{Name: quayRegistry.GetName() + "-quay-app-upgrade", Namespace: quayRegistry.GetNamespace()}, &upgradeJob)
-				if err != nil {
-					log.Error(err, "could not retrieve Quay upgrade `Job`")
-
-					return false, err
-				}
-
-				if upgradeJob.Status.Succeeded > 0 {
-					log.Info("Quay upgrade complete, updating `status.currentVersion`")
-
-					var freshQuay v1.QuayRegistry
-					if err := r.Client.Get(ctx, req.NamespacedName, &freshQuay); err != nil {
-						log.Error(err, "could not retrieve QuayRegistry")
-						return false, err
-					}
-					qcopy := freshQuay.DeepCopy()
-
-					msg := "All registry components created"
-					condition := v1.Condition{
-						Type:               v1.ConditionComponentsCreated,
-						Status:             metav1.ConditionTrue,
-						Reason:             v1.ConditionReasonComponentsCreationSuccess,
-						Message:            msg,
-						LastUpdateTime:     metav1.Now(),
-						LastTransitionTime: metav1.Now(),
-					}
-					qcopy.Status.Conditions = v1.SetCondition(qcopy.Status.Conditions, condition)
-					qcopy.Status.CurrentVersion = v1.QuayVersionCurrent
-					r.EventRecorder.Event(qcopy, corev1.EventTypeNormal, string(v1.ConditionReasonHealthChecksPassing), msg)
-
-					if err = r.Client.Status().Update(ctx, qcopy); err != nil {
-						log.Error(err, "could not update QuayRegistry status with current version")
-						return false, nil
-					}
-
-					qcopy.Spec.Components = v1.EnsureComponents(qcopy.Spec.Components)
```

Comment:
I have seen that you removed this "function" call, this is what was triggering the new event. On this case I guess that we gonna need to, once the migration is finished, return a reschedule as well.

Reply:
good catch, will fix

Reply:
done